### PR TITLE
Implemented possible values with <select> + other improvements

### PR DESCRIPTION
**Pull request by silviubogan**
- implemented possible values with `<select>` (see the examples);
- improved (documentation and non-documentation) comments and coding style (indentation, useless spaces, missing `;`);
- optimizations (`===` instead of `==`, removed useless conditions in `if`, `<th>` array appended at once);
- improved and multiplied examples;
- corrected typos.

I tested these changes and everything seems to work well. :smiley:

See #2.

### CODE REVIEW

File: src/jquery-json-editor.js [hunk 1]
```diff
@@ -342,39 +366,39 @@
...
-                         })));
-                      }
-                  }
-               }
+                var headers = [];
+                // headers
+                for (var k in field.schema) {
+                    var c = field.schema[k];
+                    headers.push(c.name);
+                    $headers.append(
```
Comment (IonicaBizau) [inline src/jquery-json-editor.js]: A small optimization: declare an array where to store the `$("<th>")` elements and then append them once: `$headers.append(yourArr)`. :balloon:

**Comment by IonicaBizau** [comment]
Tested it. Nice work! :sparkle:

File: src/jquery-json-editor.js (hunk 1 above)
Comment (silviubogan) [inline src/jquery-json-editor.js]: @IonicaBizau, done in d915cd5. :smiley:

**Comment by IonicaBizau** [comment]
:tada:

**Pull request merged**